subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java

```diff
@@ -1145,6 +1145,9 @@ private void validate(String output, String displayName) {
                         while (i < lines.size() && STACK_TRACE_ELEMENT.matcher(lines.get(i)).matches()) {
                             i++;
                         }
+                    } else if (line.matches(".*\\[deprecated.*]")) {
```

Comment:
❌ Why was this added?

Reply:
After @bmuschko 's #2880 , if we have a deprecated option, when we run `gradle --help`, we'll get:

```
USAGE: gradle [option...] [task...]

-?, -h, --help          Shows this help message.
...
--recompile-scripts     Force build script recompiling. [deprecated this has been deprecated]
...
```

Since this output contains `deprecated`, it would be treated as deprecation warning, which made many tests fail.

Reply:
Instead of adding comments, consider using a descriptive method for such conditions like `if (isDescriptionOfDeprecatedOption())`. Also consider nesting this condition inside the next one, so their order doesn't matter.

Reply:
Good advice.